extensions/steve/steve.go

```diff
@@ -0,0 +1,71 @@
+package steve
+
+import (
+	"context"
+	"errors"
+	"strings"
+	"time"
+
+	"github.com/rancher/shepherd/clients/rancher"
+	v1 "github.com/rancher/shepherd/clients/rancher/v1"
+	"github.com/sirupsen/logrus"
+	kwait "k8s.io/apimachinery/pkg/util/wait"
+)
+
+const (
+	notFound = "not found"
+)
+
+// DeleteSteveResource accepts a client, resource type and v1.SteveAPIObject and deletes the resource
+func DeleteSteveResource(client *rancher.Client, steveResourceType string, resource *v1.SteveAPIObject) error {
```

Comment:
What's the point of having this? You're just doing a delete call?

Reply:
I felt this was a nice enhancement to our extensions.  Anytime we'd like to delete a steve resource, (and this should be applied to all CRUD operations IMO) we can just call this extension, instead of repeating the client call + err != nil, which would make the tests cleaner, more readable and ultimately DRYer code, as we wouldn't be bloating tests with err != nil every time we want to make a client call.  That was my thinking in creating this.

Reply:
you still need to do a err != nil check this function returns error. Why wouldn't that happen?

Reply:
i guess it is very similar, but in the tests, having require.NoError or assert.NoError makes it more readable IMO.  

What do you want me to do about this?  I like it, but if its not getting merged, i'll nuke it 🙂

Reply:
What's the difference of doing
```
require.NoError(client.Delete)
require.NoError(DeleteSteveResource)
```
?
It really does not add much. You other functions make sense, this does not. Remove this and you got my approval.

Reply:
<img width="923" alt="Screenshot 2024-04-26 at 10 56 58 AM" src="https://github.com/rancher/shepherd/assets/46494969/b5d2301a-4fa7-4782-82bd-c51fa9bebdb4">

Reply:
updated as requested 👍